### PR TITLE
doing homework4

### PR DESCRIPTION
**Pull request by Hromy-Ann**

### CODE REVIEW

File: src/main/java/org/example/homework4/Task1.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        int year = 1889;
+        boolean leapYear = isLeapYear(year);
+        System.out.println("Year " + year + " is" + (leapYear ? "" : " not") + " leap");
+    }
+
+    private static boolean isLeapYear(int year) {
+        if (year < 1 || year > 9999) {
+            return false;
+        }
+        if (year % 4 == 0 && year % 100 != 0) {
```
Comment (alexshamrai) [inline src/main/java/org/example/homework4/Task1.java]: This is correct, but I personally prefer doing smth like this
```
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            return true;
        }
 ```
 In your case you have 2 separate expressions where return result is identical(true), but they could be combined into one

File: src/main/java/org/example/homework4/Task1.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package org.example.homework4;
+
+public class Task1 {
+    public static void main(String[] args) {
+        int year = 1889;
+        boolean leapYear = isLeapYear(year);
+        System.out.println("Year " + year + " is" + (leapYear ? "" : " not") + " leap");
```
Comment (alexshamrai) [inline src/main/java/org/example/homework4/Task1.java:7]: Lovely output)

**Review by alexshamrai — APPROVED** [review]
Great job!

File: src/main/java/org/example/homework4/Task1.java (hunk 1 above)
Comment (Hromy-Ann) [reply src/main/java/org/example/homework4/Task1.java]: Done)

**Pull request merged**